### ISSUE TITLE
Rustdoc fails, but there are no codeblocks

### ISSUE THREAD

**Issue by zmrocze** [rust-lang/rust#114025]
I encountered doctest fail on codebase that doesn't contain any codeblocks.

I guess the errors might originate from the macro expansions and code generation done by the `tonic` crate. So filed a bug report there at [BUG DESCRIPTION](https://github.com/hyperium/tonic/issues/1435). 

But it well might be a `rustdoc` bug and I suppose you guys here would know better what might be the cause, so let me cross-link the issue here as well.

### tldr

[BUG DESCRIPTION](https://github.com/hyperium/tonic/issues/1435)

**Comment by fmease** [comment]
Lacks reproduction steps.
